dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtilityTest.java

```diff
@@ -57,16 +57,25 @@ public void testReadBom_coordinates() throws ArtifactDescriptorException {
   }
 
   @Test
-  public void testReadBom_path() throws MavenRepositoryException {
+  public void testReadBom_path() throws MavenRepositoryException, ArtifactDescriptorException {
     Path pomFile = Paths.get("..", "boms", "cloud-oss-bom", "pom.xml");
     
-    Bom bom = RepositoryUtility.readBom(pomFile);
+    Bom currentBom = RepositoryUtility.readBom(pomFile);
+    Bom oldBom = RepositoryUtility.readBom("com.google.cloud:libraries-bom:2.5.0");
+    ImmutableList<Artifact> currentArtifacts = currentBom.getManagedDependencies();
+    ImmutableList<Artifact> oldArtifacts = oldBom.getManagedDependencies();
     
-    ImmutableList<Artifact> artifacts = bom.getManagedDependencies();
-    Assert.assertEquals(217, artifacts.size());
-    String coordinates = bom.getCoordinates();
+    String coordinates = currentBom.getCoordinates();
     Assert.assertTrue(coordinates.startsWith("com.google.cloud:libraries-bom:"));
     Assert.assertTrue(coordinates.endsWith("-SNAPSHOT"));
+    
+    // This is a characterization test to verify that the managed dependencies haven't changed.
+    // However sometimes this list does change. If so, we want to 
+    // output the specific difference so we can manually verify whether
+    // the changes make sense. When they do make sense, we update the test. 
+    if (currentArtifacts.size() != 217) {
```

Comment:
Is 217 the current size? Why not just always assert? If they haven't changed the test will pass.

Reply:
see above

Reply:
I'm still a bit confused. The purpose of this guard (`!= 217`) is to avoid asserting that `currentArtifacts` is the same as `oldArtifacts` unless we think the size has changed. But they could still differ even if the size _hasn't_ changed.

What you have here could induce a false negative if the actual set of dependencies changes without the size changing. Why allow that?